application/whatsapp/v0/send_message.go

```diff
@@ -0,0 +1,586 @@
+			Longitude: fmt.Sprintf("%f", inputStruct.Longitude),
+			Name:      inputStruct.LocationName,
+			Address:   inputStruct.Address,
+		},
+	}
+
+	resp, err := e.client.SendMessageAPI(&req, &TaskSendMessageResp{}, inputStruct.PhoneNumberID)
+
+	if err != nil {
+		return nil, err
+	}
+
+	respStruct := resp.(*TaskSendMessageResp)
+
+	outputStruct := TaskSendMessageOutput{
+		WaID: respStruct.Contacts[0].WaID,
+		ID:   respStruct.Messages[0].ID,
+	}
+
+	output, err := base.ConvertToStructpb(outputStruct)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert output to struct: %v", err)
+	}
+
+	return output, nil
+}
+
+// Task 4: Send Contact Message Task
+type TaskSendContactMessageInput struct {
+	PhoneNumberID   string `json:"phone-number-id"`
+	To              string `json:"to"`
+	FirstName       string `json:"first-name"`
+	MiddleName      string `json:"middle-name"`
+	LastName        string `json:"last-name"`
+	PhoneNumber     string `json:"phone-number"`
+	PhoneNumberType string `json:"phone-number-type"`
+	Email           string `json:"email"`
+	EmailType       string `json:"email-type"`
+	Birthdate       string `json:"birthday"`
+}
+
+type TaskSendContactMessageReq struct {
+	MessagingProduct string          `json:"messaging_product"`
+	To               string          `json:"to"`
+	Type             string          `json:"type"`
+	Contacts         []contactObject `json:"contacts"`
+}
+
+func (e *execution) TaskSendContactMessage(in *structpb.Struct) (*structpb.Struct, error) {
+	inputStruct := TaskSendContactMessageInput{}
+	err := base.ConvertFromStructpb(in, &inputStruct)
+
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert input to struct: %v", err)
+	}
+
+	var formattedName string
+	if inputStruct.MiddleName == "" && inputStruct.LastName == "" {
+		formattedName = inputStruct.FirstName
+	} else if inputStruct.MiddleName == "" {
+		formattedName = fmt.Sprintf("%s %s", inputStruct.FirstName, inputStruct.LastName)
+	} else if inputStruct.LastName == "" {
+		formattedName = fmt.Sprintf("%s %s", inputStruct.FirstName, inputStruct.MiddleName)
+	} else {
+		formattedName = fmt.Sprintf("%s %s %s", inputStruct.FirstName, inputStruct.MiddleName, inputStruct.LastName)
+	}
+
+	contact := contactObject{
+		Name: nameObject{
+			FormattedName: formattedName,
+			FirstName:     inputStruct.FirstName,
+			MiddleName:    inputStruct.MiddleName,
+			LastName:      inputStruct.LastName,
+		},
+		Emails: []emailObject{
+			{
+				Email: inputStruct.Email,
+				Type:  inputStruct.EmailType,
+			},
+		},
+		Birthday: inputStruct.Birthdate,
+	}
+
+	req := TaskSendContactMessageReq{
+		MessagingProduct: "whatsapp",
+		To:               inputStruct.To,
+		Type:             "contacts",
+		Contacts:         []contactObject{contact},
+	}
+
+	if inputStruct.PhoneNumber != "" {
+		if inputStruct.PhoneNumberType == "none" {
+			return nil, fmt.Errorf("you forgot to set the phone number type")
```

Comment:
nit
`you forgot to set the phone number type`
->
`please set the phone number type when you ....

Reply:
same as below